nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentModelJsonReader.java

```diff
@@ -185,6 +212,30 @@ public static void applyPropertyValues(DocumentModel src, DocumentModel dst) {
         }
     }
 
+    public static void applyAllPropertyValues(DocumentModel src, DocumentModel dst) {
+        SchemaManager service = Framework.getService(SchemaManager.class);
+        DocumentType type = service.getDocumentType(src.getType());
+        String[] schemas = type.getSchemaNames();
+        for (String schemaName : schemas) {
+            Schema schema = service.getSchema(schemaName);
+            DataModelImpl dataModel = (DataModelImpl) dst.getDataModel(schemaName);
+            DataModelImpl fromDataModel = (DataModelImpl) src.getDataModel(schemaName);
+            for (Field field : schema.getFields()) {
+                String fieldName = field.getName().getLocalName();
+                Serializable data = (Serializable) fromDataModel.getData(fieldName);
+                try {
+                    if (!(dataModel.getDocumentPart().get(fieldName) instanceof BlobProperty)) {
+                        dataModel.setData(fieldName, data);
+                    } else {
+                        dataModel.setData(fieldName, decodeBlob(data));
+                    }
```

Comment:
shouldn't that check be done recursively in case it's a sub blob property (not sure that use case is needed, though)?